### ISSUE TITLE
[PSaaS Bug]: statistics inside ignition polygons

### ISSUE THREAD

**Issue by RobBryce** [PSaaS-Developers/Project_issues#110]
### Contact Details

rbryce@heartlandsoftware.ca

### What happened?

This was first discussed in Redmine #1217

We are generating statistics for grid locations inside an ignition polygon.  These values are both possibly incorrect due to the distance from the polygon to grid locations, and meaningless because we haven't modelled anything there.  It's also computationally expensive because the algorithm continuously expands the search radius until enough vertices are located.

I'm not sure if an option is needed, or if it's appropriate to simply not provide stat's inside the ignition.

> Description
> 
> When modelling the Devils Head fire, we notice striping and patterns in various grid export statistics.  These areas were because the grid statistics were inside an ignition polygon where no vertices exist to assist in the interpolation.  This work item is to cover the work to optionally not calculate statistics inside polygon ignitions.
> 
> This work would not apply to point or line ignitions, which have no area to worry about.
> 
> If this work proceeds, then we would need an option in the API (and protobuf definition) for FGMJ files, plus likely an option on the Prometheus GUI (to assist testing and usage) on the export grid dialog.

### Version

(Linux) 2021.11.xx

### What component are you seeing the problem on?

PSaaS Binary

**Comment by spydmobile** [comment]
Agreed lets stop doing the calc.

**Comment by BadgerOnABike** [comment]
Biggest thing to test will be first timestep for point and line ignitions to ensure this initial statistic removal doesn't impact the first polygon generated.

**Comment by nealmcloughlin** [comment]
I agree with this change to stop calculating statistics for the first time step. I'd like a little more clarification for what this will look like for all outputs. I'm assuming grid statistics will not have a donut hole which reflects the initial ignition polygon. Grid statistics will only show up outside of the ignition polygon. This makes sense for statistics like HFI, ROS, and fuel consumption. What about the burn grid? Should we add another integer value to show where the ignition polygon was, i.e.) 0 = no burn, 1 = burned by simulation, 2 = ignition polygon. Area burned in the statistics table also requires some consideration and could make this a little more nuanced. If a simulation is started from a polygon, what do we report as area burned for the first time step? Do we report what the simulation burned or do we report the area of the ignition polygon plus the area burned by the simulation?

**Comment by RobBryce** [comment]
It's an easy correction for Brett's comments, to make sure that the initial polygon for a point or line ignition doesn't affect stat's.

**Comment by RobBryce** [comment]
Neal: the assumption is that for polygon ignitions, there's a donut hole where the ignition is.  It'd be easy to make an exception for (e.g.) burn grid to state it's burned, or to change the value like you indicated.  We also have a few ways to calculate a burn grid (discretization of grid cells), which would need a bit of a logic tweak to your 0, 1, 2.
We also have a mode where you can specify the start of stat's gathering, rather than assume that it's always the start of the simulation.  This was for the fuel consumption stat for NZ, but was applied to all other stat's.  This effectively creates donuts as well.

**Comment by spydmobile** [comment]
@nealmcloughlin Rob needs your response above ;-) @RobBryce tag someone with the @ key when you mention them so they know you are talking about them or asked them something ;-)
